core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/LogicalPlan.scala

```diff
@@ -204,14 +209,22 @@ case class LogicalPlan(
           else OpResultStorage.defaultStorageMode
         sink.setStorage(
           opResultStorage.create(
+            context.executionID + "_",
             storageKey,
             outputSchemaMap(o.operatorIdentifier).head,
             storageType
           )
         )
+        // add the sink collection name to the JSON array of sinks
+        sinksPointers.add(context.executionID + "_" + o.operatorID)
```

Comment:
Here you are assuming the following:
1. `storageKey == o.operatorID`
2. `collection id == context.executionID_storageKey`.
Both assumptions may not hold true in the future, I suggest either using `collectionId = xxx` and then passing it to both `opResultStorage.create` and `sinksPointers.add` to make sure they are the same thing, or having an API in `sinkStorageReader` to retrieve the collection id.

Reply:
We can refactor this in the future. For now, it works.